lib/voyager_helpers/queries.rb

```diff
@@ -165,15 +183,42 @@ def mfhd(mfhd_id)
 
       def mfhd_suppressed(mfhd_id)
         %Q(
-          SELECT suppress_in_opac 
+          SELECT suppress_in_opac
           FROM mfhd_master
           WHERE mfhd_id=#{mfhd_id}
-        ) 
+        )
+      end
+
+      def all_recap_bib_ids
+        recap_locations = '378, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 436, 438, 446, 448, 454, 459, 461, 462, 464, 465, 466, 491, 493, 494, 495, 496, 497, 498, 503, 504, 515'
+        %Q(
+          SELECT bib_master.bib_id
+          FROM (
+                 (
+                   (bib_master JOIN bib_mfhd ON bib_master.bib_id = bib_mfhd.bib_id) 
+                 JOIN mfhd_master ON bib_mfhd.mfhd_id = mfhd_master.mfhd_id) 
+               JOIN mfhd_item ON mfhd_master.mfhd_id = mfhd_item.mfhd_id) 
+          WHERE mfhd_master.location_id IN (#{recap_locations})
+          AND bib_master.suppress_in_opac = 'N'
+          AND mfhd_master.suppress_in_opac = 'N'
+          GROUP BY bib_master.bib_id
+          ORDER BY bib_master.bib_id
+        )
+      end
+
+      def recap_mfhd_ids(bib_id)
+        %Q(
+        SELECT bib_mfhd.mfhd_id
+        FROM bib_mfhd JOIN mfhd_master ON bib_mfhd.mfhd_id = mfhd_master.mfhd_id
+        WHERE (bib_id = #{bib_id}
+        and location_id IN (378, 419, 420, 421, 422, 423, 424, 425, 426, 427, 428, 429, 436, 438, 446, 448, 454, 459, 461, 462, 464, 465, 466, 491, 493, 494, 495, 496, 497, 498, 503, 504, 515)
```

Comment:
Make the `recap_locations` variable into a constant and use it here instead of listing them all twice.